src/core/components/link/link/link.tsx

```diff
@@ -34,6 +35,8 @@ export const Link = React.forwardRef<HTMLAnchorElement, ILinkProps>((props, ref)
         ...otherProps
     } = props;
 
+    const processedHref = typeof href === 'string' && isExternal ? normalizeExternalHref(href) : href;
```

Comment:
`href` is already typed as `string` here. This is the common pattern we use in these situations:

```suggestion
    const processedHref = href != null && isExternal ? normalizeExternalHref(href) : href;
```

... or even skiping the existance check and updating `normalizeExternalHref` to return `undefined` if no value. This is also a common pattern in our codebases.